miniml/compiler/compile.scm

```diff
@@ -96,9 +96,37 @@
      (TYPE typedef type_ands) : (list 'MTypedef (cons $2 $3))
      (EXCEPTION constr_decl) : (list 'MException (car $2) (cdr $2))
      (OPEN longident_uident) : (list 'MOpen $2)
-     (MODULE UIDENT EQ STRUCT list_semidefinition END) : (list 'MStruct $2 $5)
+     (MODULE UIDENT functor_args EQ module_expr) : (list 'MModule $2 $3 $5)
      (EXTERNAL lident_ext COLON type_count_arrows EQ STRING) : (list 'MExternal $2 $4 $6))
 
+   (module_expr
+    (STRUCT list_semidefinition END) : (list 'MStruct $2)
+    (longident_uident functor_apply) : (list 'MApply $1 $2))
```

Comment:
Using the prefix `M` is a bit confusing here, as it is used above for structure items. Maybe `ME`?

Reply:
I think the code would be better if you added a `FUNCTOR functor_arg MINUSGT module_expr` case here, and forced yourself to handle cleanly it below.

Reply:
Note: there is no strong need to add a production to anonymous functors of course, just using a `MEFunctor` constructor below simplifies the picture.